tiledb/sm/storage_manager/storage_manager.cc

```diff
@@ -220,8 +220,11 @@ Status StorageManagerCanonical::object_iter_begin(
   // Include the uris that are TileDB objects in the iterator state
   ObjectType obj_type;
   for (auto& uri : uris) {
-    RETURN_NOT_OK_ELSE(
-        object_type(resources_, uri, &obj_type), tdb_delete(*obj_iter));
+    try {
+      obj_type = object_type(resources_, uri);
+    } catch (...) {
+      tdb_delete(*obj_iter);
```

Comment:
```suggestion
      tdb_delete(*obj_iter);
      throw;
```

Reply:
Yes, this is needed. But it's also the case that this function is being moved out of `StorageManager` in another PR, so it's likely better to leave them all alone in this PR and remove `Status` from them after they're moved out.

Upshot: Better to revert the changes in this file.